### DISCUSSION TITLE
Under a monorepo, do we support limitations on multiple different language versions?

### DISCUSSION THREAD

**Discussion by flc1125** [renovatebot/renovate#33222] [Request Help] [answered]
### How are you running Renovate?

A Mend.io-hosted app

### Please tell us more about your question or problem

```
- go.mod // go 1.22
- package1
  - go.mod // go 1.23
- package2
  - package2.1
  - go.mod // go 1.22.10
```

As above, an example of a monorepo repository. There are multiple subpackages in it, and now I have the following configuration:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:best-practices"
  ],
  "constraints": {
    "go": "1.22"
  }
}
```

**My question is:**

- How to configure constraints to support different Go versions in the example? *(I checked the document and did not find any introduction or explanation of this kind)*
- For this [reason](https://docs.renovatebot.com/golang/#go-binary-version), in fact, it does not support the 1.22.x configuration, is my understanding correct?

**Comment by rarkins** [comment, marked as answer]
Use `packageRules` with `matchFilesNames` to have different settings per directory
